src/schema/ObjectType.ts

```diff
@@ -0,0 +1,309 @@
+   */
+  public get childObjectsInSelectedKey(): Record<string, string> {
+    const children: Record<string, string> = {};
+    this.allFields.forEach((field) => {
+      if (!this?.selectedKey?.fieldsList.includes(field.name.value)) {
+        return;
+      }
+      const fieldType = FroidSchema.extractFieldType(field);
+      if (
+        !this.objectTypes.find(
+          (searchType) => searchType.name.value === fieldType
+        )
+      ) {
+        return;
+      }
+      children[field.name.value] = fieldType;
+    });
+    return children;
+  }
+
+  /**
+   * The names of the fields that are being used by the node itself.
+   *
+   * Example schema:
+   * type Book @key(fields: "author { name }") {
+   *   author: Author!
+   * }
+   * type Author @key(fields: "authorId") {
+   *   authorId: Int!
+   *   name: String!
+   * }
+   *
+   * Example value:
+   * ['authorId']
+   *
+   * @returns {string[]} The list of field names
+   */
+  public get directlySelectedFields(): string[] {
+    return this.allFields
+      .filter((field) =>
+        this.selectedKey?.fieldsList.includes(field.name.value)
+      )
+      .map((field) => field.name.value);
+  }
+
+  /**
+   * The names of the fields that are referenced in another entity's key.
+   *
+   * Example schema:
+   * type Book @key(fields: "author { name }") {
+   *   author: Author!
+   * }
+   * type Author {
+   *   name: String!
+   * }
+   *
+   * Example value:
+   * ['name']
+   *
+   * @returns {string[]} The list of field names
+   */
+  public get externallySelectedFields(): string[] {
+    return this._externallySelectedFields;
+  }
+
+  /**
+   * The list of all fields referenced in the node key and by other entities.
+   *
+   * @returns {FieldDefinitionNode} The list of fields
+   */
+  public get selectedFields(): FieldDefinitionNode[] {
+    return this.allFields.filter(
+      (field) =>
+        this.directlySelectedFields.includes(field.name.value) ||
+        this.externallySelectedFields.includes(field.name.value)
+    );
+  }
+
+  /**
+   * The list of selected fields that appear in any of the node's keys.
+   *
+   * @returns {FieldDefinitionNode[]} The list of key fields
+   */
+  public get selectedKeyFields(): FieldDefinitionNode[] {
+    return this.selectedFields.filter((field) =>
+      this.allKeyFieldsList.includes(field.name.value)
+    );
+  }
+
+  /**
+   * The list of selected fields that do not appear in any of the node's keys.
+   *
+   * @returns {FieldDefinitionNode[]} The list of non-key fields
+   */
+  public get selectedNonKeyFields(): FieldDefinitionNode[] {
+    return this.selectedFields.filter(
+      (field) => !this.allKeyFieldsList.includes(field.name.value)
+    );
+  }
+
+  /**
+   * The node's key after all key fields used by other entities are added.
+   *
+   * @todo handle key recursion:
+   * - Have a max depth and kill recursion if it reaches the max depth (100 levels? 1000 levels?)
```

Comment:
I think 100 is a sane start?

Reply:
I think I went with 200 (and forgot to remove this todo) but 100 is likely more than enough. I could probably make it 10 or 20, honestly. The goal here is to make an escape hatch to prevent max call stack errors.

Reply:
I had gone with 200. I updated it to 100. Let me know if you think I should go lower.